src/connectivity-api/connectivity-qt/connectivityqt/connectivity.cpp

```diff
@@ -387,6 +390,34 @@ void Connectivity::setHotspotEnabled(bool enabled)
 
     // TODO Remove this when SDK switch widget isn't broken
     d->m_propertyCache->setProperty("HotspotEnabled", enabled);
+
+    // HACK: For switching MediaTek /dev/wmtWifi adapter mode between AP/STA
+    //       when hotspot enabled state is changed.
+    try
+    {
+        const char* wmtWifi_dev = "/dev/wmtWifi";
+
+        QFileInfo wmtWifi_fileinfo(wmtWifi_dev);
+        if (wmtWifi_fileinfo.exists())
+        {
+            QFile wmtWifi_file(wmtWifi_dev);
+            wmtWifi_file.open(QIODevice::WriteOnly);
+
+            // Reset adapter state
+            const char* resetAdapterMode = "0";
+            wmtWifi_file.write(resetAdapterMode, qstrlen(resetAdapterMode));
```

Comment:
Write returns -1 on failure, add a way to please handle this (logging it would be fine) 

https://doc.qt.io/qt-5/qiodevice.html#write